### PR TITLE
MM-47288 refactor: convert channels/channel/ files to ts

### PR DESCRIPTION
**Pull request by hereje**
#### Summary

refactor: convert channels/channel/ files to ts

- convert js files to typescript
- update types
- fix lint issues

#### Ticket Link

Fixes: #21299

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by larkox** [comment]
@hereje It seems like there are some lint issues. Can you take a look?

**Comment by hereje** [comment]
> @hereje It seems like there are some lint issues. Can you take a look?

Sure!

File: e2e-tests/cypress/tests/support/api/user.d.ts [hunk 1]
```diff
@@ -248,7 +248,7 @@ declare namespace Cypress {
          *       // do something with users
          *   });
          */
-        apiGetUsers(queryParams: Record<string, any>): Chainable<UserProfile[]>;
+        apiGetUsers(queryParams: Record<string, any>): Chainable<{users: UserProfile[]}>;
```
Comment (pvev) [inline e2e-tests/cypress/tests/support/api/user.d.ts:251]: 👍

File: e2e-tests/cypress/tests/support/external_commands.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ import {getRandomId} from '../utils';
 function externalActivateUser(userId: string, active = true) {
     const admin = getAdminAccount();
 
-    cy.externalRequest({user: admin, method: 'PUT', path: `users/${userId}/active`, data: {active}});
+    return cy.externalRequest({user: admin, method: 'PUT', path: `users/${userId}/active`, data: {active}});
```
Comment (pvev) [inline e2e-tests/cypress/tests/support/external_commands.ts:16]: any special reason for this return here?

**Review by pvev — APPROVED** [review]
LGTM! just a couple of questions but in general this looks good. Thanks @hereje 🚀

**Review by larkox — APPROVED** [review]
No extra comments from my side

File: e2e-tests/cypress/tests/support/external_commands.ts (hunk 2 above)
Comment (hereje) [reply e2e-tests/cypress/tests/support/external_commands.ts:16]: So the function does not break the functionality in `channels/enterprise/guest_accounts/guest_identification_ui_not_cloud_spec.ts`

```typescript
// # Deactivate Guest user
cy.externalActivateUser(guestUser.id, false).wait(TIMEOUTS.FIVE_SEC); // .wait
```

File: e2e-tests/cypress/tests/integration/channels/channel/archived_channels_2_spec.ts [hunk 3]
```diff
@@ -45,12 +48,12 @@ describe('Leave an archived channel', () => {
...
                 team_id: testTeam.id,
                 type: 'P',
-            })).then(async (channel) => {
+            } as Channel)).then(async (channel: ServerChannel) => {
                 // # Then invite us to it
                 await client.addToChannel(testUser.id, channel.id);
 
-                cy.wrap(channel);
-            }).then((channel) => {
+                return Promise.resolve(channel);
```
[the remark this answers is not in the document]
Comment (hereje) [reply e2e-tests/cypress/tests/integration/channels/channel/archived_channels_2_spec.ts]: I've made some changes. I would like to know your comments.

**Comment by saturninoabril** [comment]
Thanks @hereje for your contribution!

I just ran the tests to make sure all working as expected and nothing is broken. Now ready to merge (set to auto-merge).

**Pull request merged**